### ISSUE TITLE
`fsdp_sft_trainer.py`: question on handling not-finite `grad_norm`

### ISSUE THREAD

**Issue by vadimkantorov** [volcengine/verl#1506]
https://github.com/volcengine/verl/blob/52437be1a61336f69b79a0540e5b89c9e93dbffd/verl/trainer/fsdp_sft_trainer.py#L386-L390

```
        # if grad_norm is not finite, skip the update
        if not torch.isfinite(grad_norm):
            print(f"WARN: grad_norm is not finite: {grad_norm}")
            self.optimizer.zero_grad()
        else:
            self.optimizer.step()
```

Could this cause divergence between ranks? Or is it not happening because grad_norm is computer after all gradients are already synchronized and `grad_norm` is exactly the same between all ranks?

Thanks!

**Comment by Kirrito-k423** [comment]
定时清理检查结论：该 issue 创建于 2025-05-13T12:23:10Z，截至 2026-07-06 已超过 6 个月；comments 数量为 0；timeline 检查未发现公开引用、PR/issue cross-reference、commit/PR mention 或其它公开关联互动。

检查来源：issue 页面 https://github.com/verl-project/verl/issues/1506，以及该 issue 的公开 comments/timeline 记录。根据本次定时清理规则，创建超过 6 个月且零评论零公开引用的 open issue 将先关闭。如果作者或社区成员仍能提供新的复现、证据或需求背景，请继续评论请求重开，或创建新的 issue 并引用这里的上下文。

---

Scheduled cleanup result: this issue was created at 2025-05-13T12:23:10Z, which is more than 6 months before 2026-07-06; it has 0 comments; the timeline check found no public references, PR/issue cross-references, commit/PR mentions, or other public linked interactions.

Checked sources: the issue page https://github.com/verl-project/verl/issues/1506, plus this issue's public comments/timeline records. Under the scheduled cleanup rule, open issues older than 6 months with zero comments and zero public references are closed. If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue and link this context.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
